src/service.ts

```diff
@@ -319,6 +321,15 @@ export class Service implements IService {
     return utils.find(x => x.config.uniqueId === uniqueId, this.embeds);
   }
 
+  addOrOverwriteEmbed(component: embed.Embed, element: HTMLElement): void {
+    // remove embeds over the same div element.
+    this.embeds = this.embeds.filter(function(embed) {
```

Comment:
Why aren't you removing the identical component before re-appending it? Each embed.load( will leave an orphaned component?